src/Controls/tests/Core.UnitTests/ShellLifeCycleTests.cs

```diff
@@ -541,6 +544,123 @@ public async Task OnNavigatedFiresWhenPopToRoot()
 			shell.TestCount(1);
 		}
 
+		[Fact]
+		public async Task VerifyPageAppearingSequence()
+		{
+			var shell = new TestShell();
+			Routing.RegisterRoute("ShellLifeCycleTestPage1", typeof(ShellLifeCycleTestPage1));
+
+			int appearingCountHomePage = 0;
+			shell.FlyoutBehavior = FlyoutBehavior.Disabled;
+			var mainContent = new ShellContent
+			{
+				Title = "Home",
+				ContentTemplate = new DataTemplate(() => 
+				{
+					var homePage =  new ShellLifeCycleTestHomePage();
+					homePage.Appearing += (s, e) => 
+					{
+						appearingCountHomePage++;
+					};
+					return homePage;
+				}),
+				Route = "ShellLifeCycleTestHomePage",
+			};
+
+			var tab = new Tab { Title = "TestTab" };
+			tab.Items.Add(new ShellContent
+			{
+				Title = "Parameter",
+				Route = "ShellLifeCycleTestPage2",
+				ContentTemplate = new DataTemplate(() => new ShellLifeCycleTestPage2())
+			});
+
+			shell.Items.Add(mainContent);
+			var tabbar = new TabBar();
+			tabbar.Items.Add(tab);
+			shell.Items.Add(tabbar);
+			int firstPageAppearingCount = 1;
+			await shell.GoToAsync("ShellLifeCycleTestPage1");
+			(shell.CurrentPage as ShellLifeCycleTestPage1).Appearing += (s, e) =>
+			{
+				firstPageAppearingCount++;
+			};
+
+			await shell.GoToAsync("//ShellLifeCycleTestPage2");
+			await shell.GoToAsync("//ShellLifeCycleTestHomePage");
+			Assert.Equal(1, firstPageAppearingCount);
+			Assert.Equal(2, appearingCountHomePage);
+		}
+
+
+
+		[Fact]
+		public async Task VerifyPageAppearingSequence2() // Temporary Name
```

Comment:
Fixed the shell content so if it's created as a template it creates correctly and life cycle events all fire.

I also added this test to demonstrate an issue with the code added for this fixed. 

I think the fix for this one is going to need to be moved into `PresentedPageAppearing`